src/helpers/node/config.js

```diff
@@ -1,64 +1,87 @@
 /**
  * Configuration file for node runtime
- * TODO?: Remove all side-effects and use a constructor?
- * - This module has side effects, sequentially setting up the environment.
- * - Only variables may be exported from this module.
- * - Don't define functions here, import functions if required.
+ * TODO: Remove all side-effects and use a constructor?
+ * This module has side effects, sequentially setting up the environment.
  */
 
 import { atob, btoa } from "buffer";
 import fetch, { Headers, Request, Response } from "node-fetch";
 import { getTLSfromEnv } from "./util.js";
-import EnvManager from "../env.js";
 import Log from "../log.js";
+import * as system from "../../system.js";
+import EnvManager from "../env.js";
+import * as swap from "../linux/swap.js";
+
+(async (main) => {
+  // if this file execs... assume we're on nodejs.
+  const runtime = "node";
+  const isProd = process.env.NODE_ENV === "production";
+  const onFly = process.env.CLOUD_PLATFORM === "fly";
+  let devutils = null;
+  let dotenv = null;
+
+  // dev utilities
+  if (!isProd) {
+    devutils = await import("./util-dev.js");
+    dotenv = await import("dotenv");
+  }
+
+  /** Environment Variables */
+  // Load env variables from .env file to process.env (if file exists)
+  // NOTE: this won't overwrite existing
+  if (dotenv) {
+    dotenv.config();
+    console.log("loading local .env");
+  }
+
+  console.log("override runtime, from", process.env.RUNTIME, "to", runtime);
+  process.env.RUNTIME = runtime; // must call before creating env-manager
 
-/** Environment Variables */
-// Load env variables from .env file to process.env (if file exists)
-// NOTE: this won't overwrite existing
-if (process.env.NODE_ENV !== "production") (await import("dotenv")).config();
-process.env.RUNTIME = "node";
+  globalThis.envManager = new EnvManager();
 
-if (!globalThis.envManager) globalThis.envManager = new EnvManager();
-envManager.loadEnv();
+  /** Logger */
+  globalThis.log = new Log(
+    env.logLevel,
+    isProd // set console level only in prod.
+  );
 
-/** Logging level */
-globalThis.log = new Log(
-  env.logLevel,
+  /** TLS crt and key */
+  const _TLS_CRT_KEY =
+    eval(`process.env.TLS_${process.env.TLS_CN}`) || process.env.TLS_;
 
-  // set console level only in production
-  !console.level && env.runTimeEnv === "production"
-);
+  const [TLS_KEY, TLS_CRT] =
+    isProd || _TLS_CRT_KEY !== undefined
+      ? getTLSfromEnv(_TLS_CRT_KEY)
+      : devutils.getTLSfromFile(
+          process.env.TLS_KEY_PATH,
+          process.env.TLS_CRT_PATH
+        );
+  envManager.set("tlsKey", TLS_KEY);
+  envManager.set("tlsCrt", TLS_CRT);
```

Comment:
This is not needed in globally available in `env` !?

Reply:
I intend to drop global `env`, and move everything behind `envMap`, in a subsequent commit.

Reply:
`env` replaced with `envManager` now: https://github.com/serverless-dns/serverless-dns/commit/b469d5239325771d203f94fce0da9716242c71a1#diff-36b6afa29e4f1137c0357af87a6fe4199e9023ae3339ef2aa85619559268ac78